File: project-program2 - Copy/core/src/com/mygdx/game/PhysicsImp.java
package com.mygdx.game;

public class PhysicsImp {
    // setting up the screen variables
    public static final int S_WIDTH = 1350; // screen width
    public static final int S_HEIGHT = 759; // screeen height
    public static final int W_WIDTH = 10560 * 16; // world width
    public static final int W_HEIGHT = 50 * 16; // world height

    // setting up physical variables
    public static final float AIR_DENSITY = 1.225f;

    //bomb properties
    public static final int BOMB_DENSITY = 2181;
    public static final int BOMB_LENGTH = 1;
    //changable properties
    private int radius; // setting the radius to a public static int as this will allow for it to be changed in the main menu
    private int bombRPM;
    private int planeSpeed ;
    private int criticalAngle;

    //sim properties
    private int startDistance = 12000;
    private float unitScale = 20; // this will scale up all our vector quantities to increase simulation speeds to something more realistic
    private boolean planeFLyAway = false, bombSinks = false, damDestroyed = false, bombHitsWater = false, bombDamaged = false;

    //for destroying layers to enable the bomb to sink or the dam to be destroyed
    public static final short DEFAULT_BIT = 1;
    public static final short BOMB_BIT = 2;
    public static short WATER_BIT = 4;
    public static final short DAM_BIT = 8;
    public static final short DESTROYED_BIT = 16;
    public static final short EXPLOSION_BIT = 32;



    public static float WEIGHT_OF_BOMB(int radius, int density, int length){
        float weight = MASS_OF_BOMB(radius, density, length) * 10;
        return weight;
    }

    public static float MASS_OF_BOMB(int radius, int density, int length){
        float mass = (float) (Math.PI * length * density * (radius*radius));
        return mass;
    }

    public static float LIFT_FORCE(int radius, int RPM, float velocityX ){
        //return vortex(bomb.getRotationalVelocity(), bomb.radius) * AIR_DENSITY * bomb.b2dbody.getLinearVelocity().x;
        return VORTEX(radius, RPM) * AIR_DENSITY * velocityX;
    }

    public static float VORTEX(int radius, float RPM){
        float vortexStrength = (float) (2 * AIR_DENSITY * (RPM  * 0.10472f) * (float)(radius * radius)) ;
        return (float) (vortexStrength);
    }

    public static float TOTAL_DOWNWARD_FORCE(float liftForce, float downForce){
        return (downForce - liftForce);
    }

    public static float TOTAL_ACCELERATION(float totalDownForce, float mass){
        return ((totalDownForce /mass));
    }

    public static float CRITICAL_ANGLE(float acceleration){
        return (float) (18/(Math.sqrt(acceleration)));
    }

    public static float ANGLE_OF_INCIDENCE(float x_vel, float y_vel){
        float angleOfIncidence = Math.abs((float) Math.atan (y_vel/x_vel) * 180/ (float)Math.PI);
        return angleOfIncidence;
    }

    public static boolean WillItBounce(float angleOfIncidence, float citicalAngle){
        if (angleOfIncidence < citicalAngle){
            return true;
        }
        else{
            return  false;
        }
    }

    public int getRadius() {
        return radius;
    }

    public void setRadius(int radius) {
        this.radius = radius;
    }

    public int getBombRPM() {
        return bombRPM;
    }

    public void setBombRPM(int bombRPM) {
        this.bombRPM = bombRPM;
    }

    public int getPlaneSpeed() {
        return planeSpeed;
    }

    public void setPlaneSpeed(int planeSpeed) {
        this.planeSpeed = planeSpeed;
    }

    public int getCriticalAngle() {
        return criticalAngle;
    }

    public void setCriticalAngle(int criticalAngle) {
        this.criticalAngle = criticalAngle;
    }

    public int getStartDistance() {
        return startDistance;
    }

    public void setStartDistance(int startDistance) {
        this.startDistance = startDistance;
    }

    public float getUnitScale() {
        return unitScale;
    }

    public void setUnitScale(float unitScale) {
        this.unitScale = unitScale;
    }

    public boolean isPlaneFLyAway() {
        return planeFLyAway;
    }

    public void setPlaneFLyAway(boolean planeFLyAway) {
        this.planeFLyAway = planeFLyAway;
    }

    public boolean isBombSinks() {
        return bombSinks;
    }

    public void setBombSinks(boolean bombSinks) {
        this.bombSinks = bombSinks;
    }

    public boolean isDamDestroyed() {
        return damDestroyed;
    }

    public void setDamDestroyed(boolean damDestroyed) {
        this.damDestroyed = damDestroyed;
    }

    public boolean isBombHitsWater() {
        return bombHitsWater;
    }

    public void setBombHitsWater(boolean bombHitsWater) {
        this.bombHitsWater = bombHitsWater;
    }

    public boolean isBombDamaged() {
        return bombDamaged;
    }

    public void setBombDamaged(boolean bombDamaged) {
        this.bombDamaged = bombDamaged;
    }
}

File: project-program2 - kotlin/core/src/com/mygdx/game/PhysicsImp.kt
package com.mygdx.game

import com.mygdx.game.PhysicsImp

object PhysicsImp {
    // setting up the screen variables
    const val S_WIDTH = 1350 // screen width
    const val S_HEIGHT = 759 // screeen height
    const val W_WIDTH = 10560 * 16 // world width
    const val W_HEIGHT = 50 * 16 // world height

    // setting up physical variables
    const val AIR_DENSITY = 1.225f

    //bomb properties
    const val BOMB_DENSITY = 2181
    const val BOMB_LENGTH = 1
    @JvmField
    var RADIUS =
        12 // setting the radius to a public static int as this will allow for it to be changed in the main menu
    @JvmField
    var BOMB_RPM = 600
    @JvmField
    var PLANE_SPEED = 80
    var CRITICAL_ANGLE = 16

    //sim properties
    @JvmField
    var START_DISTANCE = 12000
    @JvmField
    var UNITSCALE =
        20f // this will scale up all our vector quantities to increase simulation speeds to something more realistic
    @JvmField
    var PLANE_FLY_AWAY = false
    @JvmField
    var BOMB_SINKS = false
    @JvmField
    var DAM_DESTROYED = false
    @JvmField
    var BOMB_HITS_WATER = false
    @JvmField
    var BOMB_DAMAGED = false

    //for destroying layers to enable the bomb to sink or the dam to be destroyed
    const val DEFAULT_BIT: Short = 1
    const val BOMB_BIT: Short = 2
    @JvmField
    var WATER_BIT: Short = 4
    const val DAM_BIT: Short = 8
    const val DESTROYED_BIT: Short = 16
    const val EXPLOSION_BIT: Short = 32
    @JvmStatic
    fun WEIGHT_OF_BOMB(
        radius: Int,
        density: Int,
        length: Int
    ): Float {
        return MASS_OF_BOMB(radius, density, length) * 10
    }

    @JvmStatic
    fun MASS_OF_BOMB(radius: Int, density: Int, length: Int): Float {
        return (Math.PI * length * density * (radius * radius)).toFloat()
    }

    @JvmStatic
    fun LIFT_FORCE(radius: Int, RPM: Int, velocityX: Float): Float {
        //return vortex(bomb.getRotationalVelocity(), bomb.radius) * AIR_DENSITY * bomb.b2dbody.getLinearVelocity().x;
        return VORTEX(radius, RPM.toFloat()) * AIR_DENSITY * velocityX
    }

    fun VORTEX(radius: Int, RPM: Float): Float {
        return (2 * AIR_DENSITY * (RPM * 0.10472f) * (radius * radius).toFloat())
    }

    @JvmStatic
    fun TOTAL_DOWNWARD_FORCE(liftForce: Float, downForce: Float): Float {
        return downForce - liftForce
    }

    @JvmStatic
    fun TOTAL_ACCELERATION(totalDownForce: Float, mass: Float): Float {
        return totalDownForce / mass
    }


    @JvmStatic
    fun ANGLE_OF_INCIDENCE(x_vel: Float, y_vel: Float): Float {
        return Math.abs(
            Math.atan((y_vel / x_vel).toDouble()).toFloat() * 180 / Math.PI.toFloat()
        )
    }

    @JvmStatic
    fun WillItBounce(angleOfIncidence: Float): Boolean {
        return if (angleOfIncidence < CRITICAL_ANGLE) {
            true
        } else {
            false
        }
    }
}